script/DeployAll.s.sol

```diff
@@ -2,13 +2,14 @@
 pragma solidity ^0.8.4;
 
 import {Script} from "forge-std/Script.sol";
-import "../src/Delegation.sol";
+import "../src/DeployAll.sol";
 
-contract DeployDelegationImplementationScript is Script {
+contract DeployAllScript is Script {
     function run() external {
         uint256 deployerPrivateKey = vm.envUint("PRIVATE_KEY");
+        address deployer = vm.createWallet(deployerPrivateKey).addr;
```

Comment:
it's an anti-pattern to be doing this stuff IMO and you should be using `vm.startBroadcast()` and rely on foundry to inject the right signer here, either via `--private-key` or via `--aws` etc.